ros2bag/ros2bag/verb/play.py

```diff
@@ -208,15 +208,17 @@ def main(self, *, args):  # noqa: D102
 
         play_options.exclude_regex_to_filter = args.exclude_regex
 
-        play_options.exclude_topics_to_filter = args.exclude_topics if args.exclude_topics else []
-
         play_options.exclude_service_events_to_filter = \
             convert_service_to_service_event_topic(args.exclude_services)
 
         play_options.topic_qos_profile_overrides = qos_profile_overrides
         play_options.loop = args.loop
         play_options.topic_remapping_options = topic_remapping
         play_options.clock_publish_frequency = args.clock
+        exclude_topics = args.exclude_topics if args.exclude_topics else []
+        if play_options.clock_publish_frequency > 0:
+            exclude_topics.append('/clock')
+        play_options.exclude_topics_to_filter = exclude_topics
```

Comment:
Can we just do the following without any other changes above?

```suggestion
        if play_options.clock_publish_frequency > 0:
            play_options.exclude_topics_to_filter.append('/clock')
```

Reply:
I haven't been able to trace the cause in detail, but I couldn't add it directly :thinking: 

```
print(f"play_options.clock_publish_frequency: {play_options.clock_publish_frequency}")
if play_options.clock_publish_frequency > 0:
    print("Before appending:", play_options.exclude_topics_to_filter)
    play_options.exclude_topics_to_filter.append('/clock')
    print("After appending:", play_options.exclude_topics_to_filter)
print(f"Final state of exclude_topics_to_filter: {play_options.exclude_topics_to_filter}")
```

```
play_options.clock_publish_frequency: 100.0
Before appending: []
After appending: []
Final state of exclude_topics_to_filter: []
```